pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -126,7 +126,7 @@ func makeVirtualServiceSpec(ia v1alpha1.IngressAccessor, gateways map[v1alpha1.I
 	gw.Insert(gateways[v1alpha1.IngressVisibilityClusterLocal]...)
 	gw.Insert(gateways[v1alpha1.IngressVisibilityExternalIP]...)
 	spec := v1alpha3.VirtualServiceSpec{
-		Gateways: gw.List(),
+		Gateways: qualifyGateways(gw.List()),
```

Comment:
Why not simply update the ConfigMap and the default gateways?

Reply:
Because here, it is assuming that the gateway is inside Knative namespace.
It is not very future proof.

Reply:
Yes, we should update the ConfigMap seperately, but this change will still be needed to be backward compat with existing ConfigMaps (which was why leave the qualified ones alone)